plugins/libplugin-pay.c

```diff
@@ -78,6 +78,41 @@ struct payment *payment_root(struct payment *p)
 		return payment_root(p->parent);
 }
 
+static void
+paymod_log(struct payment *p, enum log_level l, const char *fmt, ...)
+{
+	struct payment *root = payment_root(p);
+	char *txt;
+	va_list ap;
+
+	va_start(ap, fmt);
+	txt = tal_vfmt(tmpctx, fmt, ap);
+	va_end(ap);
+
+	plugin_log(p->plugin, l,
+		   "id %"PRIu64" partid %"PRIu32": %s",
+		   (root->cmd && root->cmd->id) ? *root->cmd->id : UINT64_MAX,
+		   p->partid,
+		   txt);
```

Comment:
Printing `UINT64_MAX` seems very hacky to me. Could we add a `(null)` to indicate that this was not initiated by `cmd` or have a global counter of payments and just increment that for each `payment_new` we call?

Reply:
The `id` is supposed to identify the command that initiated the call, so that e.g. if we have two commands in parallel, we do no get confused which log belongs to which command.  Using the `cmd` id is better since it correlates with a command coming from `lightningd`, though of course `lightningd` never prints that.

Reply:
Modified to print `cmd N` if there is a command with an id, fall back to `id N` and a generated ID if not exist.